src/Tasks/Microsoft.NET.Build.Tasks/PrepareForReadyToRunCompilation.cs

```diff
@@ -400,28 +442,47 @@ private bool GetCrossgenComponentsPaths()
             {
                 if (_targetArchitecture == Architecture.Arm)
                 {
-                    _crossgenPath = Path.Combine(_packagePath, "tools", "x86_arm", "crossgen.exe");
-                    _clrjitPath = Path.Combine(_packagePath, "runtimes", "x86_arm", "native", "clrjit.dll");
-                    _diasymreaderPath = Path.Combine(_packagePath, "runtimes", _runtimeIdentifier, "native", "Microsoft.DiaSymReader.Native.x86.dll");
+                    if (RuntimeInformation.OSArchitecture == _targetArchitecture)
```

Comment:
This is a fix to a separate bug... maybe it should go into a separate PR? Let me know

Reply:
What's the bug? Yes, separate PR. Could be helpful if we need to port to other branches?

Reply:
The bug is that we couldn't run the native arm/arm64 crossgen and publish from a Windows arm/arm64 device. I'll move that to a separate PR.